### PR TITLE
[new] added the extended advertisement contracts

### PR DESCRIPTION
**Pull request by GustavoRSSilva**
## AppCoins Protocol smart contracts

Thank you for contributing! Please take a moment to review our [**contributing guidelines**](https://github.com/AppStoreFoundation/asf-contracts/wiki/Contributing-to-AppCoins-Protocol-smart-contracts)
to make the process easy and effective for everyone involved.

**Please open an issue** before embarking on any significant pull request, otherwise you risk spending a lot of time working
on something that might not end up being merged into the project.

Before opening a pull request, please ensure:

- [x] You have followed our [**contributing guidelines**](https://github.com/AppStoreFoundation/asf-contracts/wiki/Contributing-to-AppCoins-Protocol-smart-contracts)
- [x] double-check your branch is based on `dev` and targets `dev` 
- [x] Pull request has tests (at least 80% coverage)
- [x] Code is well-commented, linted and follows project conventions
- [x] Documentation is updated (if necessary)
- [x] Description explains the issue/use-case resolved and auto-closes related issues

Be kind to code reviewers, please try to keep pull requests as small and focused as possible :)

**IMPORTANT**: By submitting a patch, you agree to allow the project
owners to license your work under the terms of the [GPL-3.0 License](https://github.com/AppStoreFoundation/asf-contracts/blob/master/LICENSE).

### CODE REVIEW

File: scripts/extended-advertisement-storage-finance.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+const ExtendedAdvertisementStorage = artifacts.require("./ExtendedAdvertisementStorage.sol");
```
Comment (DioPires) [inline scripts/extended-advertisement-storage-finance.js:1]: #### Tests:
```diff 
- 4 failing
```
```diff 
+ 100 passing
```
#### Test coverage: 
81.44 %

File: scripts/extended-advertisement-storage-finance.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+module.exports = function(callback) {
+    let extendedStorageInstance;
+    let extendedFinanceInstance;
+    let extendedAdvertisementStorageAddress;
+
+    switch (network) {
+
+        case 'development':
+            extendedStorageInstance = ExtendedExtendedAdvertisementStorage.at(process.env.EXTENDED_ADVERTISEMENT_STORAGE_DEVELOPMENT_ADDRESS);
+            extendedFinanceInstance = ExtendedExtendedAdvertisementFinance.at(process.env.EXTENDED_ADVERTISEMENT_FINANCE_DEVELOPMENT_ADDRESS);
```
Comment (tcrosado) [inline scripts/extended-advertisement-storage-finance.js]: I think there is a typo in this  two lines

File: scripts/extended-advertisement-storage-finance.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+
+    const addAdvertisementContractAddressToFinance = function(newAdvertisementAddress) {
+        extendedFinanceInstance.setAllowedAddress(newAdvertisementAddress).then(function(error, success) {
+            console.log(`New Extended Advertisement address ${newAdvertisementAddress} added to the finance!`);
+        })
+    }
+
+    const addAdvertisementStorageContractAddressToFinance = function(newAdvertisementAddress) {
+        extendedFinanceInstance.setAdsStorageAddress(newAdvertisementAddress).then(function(error, success) {
```
Comment (tcrosado) [inline scripts/extended-advertisement-storage-finance.js]: Shouldn't this variable be a newAdvertisementStorageAddress instead?

**Review by tcrosado — CHANGES_REQUESTED** [review]

File: scripts/extended-advertisement-storage-finance.js (hunk 1 above)
Comment (DioPires) [inline scripts/extended-advertisement-storage-finance.js:1]: #### Tests:
```diff 
- 4 failing
```
```diff 
+ 100 passing
```
#### Test coverage: 
81.44 %

Comment (DioPires) [inline scripts/extended-advertisement-storage-finance.js:1]: #### Tests:
```diff 
- 4 failing
```
```diff 
+ 100 passing
```
#### Test coverage: 
81.44 %

Comment (DioPires) [inline scripts/extended-advertisement-storage-finance.js:1]: #### Tests:
```diff 
- 4 failing
```
```diff 
+ 100 passing
```
#### Test coverage: 
81.44 %

Comment (DioPires) [inline scripts/extended-advertisement-storage-finance.js:1]: #### Tests:
```diff 
- 4 failing
```
```diff 
+ 100 passing
```
#### Test coverage: 
81.44 %

**Review by tcrosado — APPROVED** [review]

File: scripts/extended-advertisement-storage-finance.js (hunk 1 above)
Comment (DioPires) [inline scripts/extended-advertisement-storage-finance.js:1]: #### Tests:
```diff 
- 4 failing
```
```diff 
+ 100 passing
```
#### Test coverage: 
81.44 %

**Pull request merged**